paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonTheme.kt

```diff
@@ -0,0 +1,173 @@
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.isUnspecified
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.isUnspecified
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.uicore.getBackgroundColor
+import com.stripe.android.uicore.getBorderStrokeColor
+import com.stripe.android.uicore.getOnBackgroundColor
+
+internal data class PrimaryButtonColors(
+    val background: Color = Color.Unspecified,
+    val onBackground: Color = Color.Unspecified,
+    val successBackground: Color = Color.Unspecified,
+    val onSuccessBackground: Color = Color.Unspecified,
+    val border: Color = Color.Unspecified,
+)
+
+internal data class PrimaryButtonShape(
+    val cornerRadius: Dp = Dp.Unspecified,
+    val borderStrokeWidth: Dp = Dp.Unspecified
+)
+
+internal data class PrimaryButtonTypography(
+    val fontFamily: FontFamily? = null,
+    val fontSize: TextUnit = TextUnit.Unspecified,
+)
+
+internal val LocalPrimaryButtonColors = staticCompositionLocalOf {
+    PrimaryButtonColors()
+}
+
+internal val LocalPrimaryButtonShape = staticCompositionLocalOf {
+    PrimaryButtonShape()
+}
+
+internal val LocalPrimaryButtonTypography = staticCompositionLocalOf {
+    PrimaryButtonTypography()
+}
+
+internal object PrimaryButtonTheme {
+    val colors: PrimaryButtonColors
+        @Composable
+        get() = getPrimaryButtonColors()
+
+    val shape: PrimaryButtonShape
+        @Composable
+        get() = getPrimaryButtonShape()
+
+    val typography: PrimaryButtonTypography
+        @Composable
+        get() = getPrimaryButtonTypography()
+
+    @Composable
+    private fun getPrimaryButtonColors(): PrimaryButtonColors {
+        val style = StripeTheme.primaryButtonStyle
+        val context = LocalContext.current
+        val localColors = LocalPrimaryButtonColors.current
+        val isDarkTheme = isSystemInDarkTheme()
+
+        val defaultSuccessBackground =
+            colorResource(id = R.color.stripe_paymentsheet_primary_button_success_background)
+
+        return remember(
+            style,
+            context,
+            localColors,
+            isDarkTheme
+        ) {
+            PrimaryButtonColors(
+                background = if (localColors.background.isUnspecified) {
+                    Color(style.getBackgroundColor(context))
+                } else {
+                    localColors.background
```

Comment:
Should we just make these nullable and do `localColors.background ?: <default>`?

Reply:
I'd prefer we keep as much non-nullable as possible in the theme. That being said, these `if` statements do make the code unreadable. `Color`, `Dp`, & `TextUnit` do have an extension function `takeOrElse` which does this internally that I have updated the PR with.

Reply:
The benefit of nullability is that we’re forced to deal with it by the compiler, which isn’t the case for a non-nullable default value. Happy to move this chat elsewhere though to not hold this up.